File: bubble_sort/bubbleSort.js
const bubbleSort = array => {
	for (let i = 0; i < array.length; i++) {
		let swap = false;
		for (let j = 0; j < array.length-i; j++) {
			if (array[j]>array[j+1]) {
				let temp = array[j];
				array[j] = array[j+1];
				array[j+1]=temp;
				swap = true;
			}
		}
		if (swap===false) {
			break;
		}
	}
	return array
}

console.log(bubbleSort([5, 4, 1, 8, 2]));

File: bubble_sort/bubble_sort.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <sstream>
#include <vector>

using namespace std;

vector<int> bubble_sort(vector<int> array);

int main(){
	int n;
	cin>>n;
	vector<int> a(n);
	for (int i = 0; i<n; i++) {
		cin>>a[i];
	}
	vector<int> sorted = bubble_sort(a);
	for(int value : sorted){
		cout<<value<<" ";
	}
	cout<<endl;
	return 0;
}

vector<int> bubble_sort(vector<int> array){
	bool swapped;
	for (int i=0; i<array.size()-1; i++) {
		swapped = false;
		for (int j=0; j<array.size()-i-1; j++) {
			if(array[j]>array[j+1]){
				swap(array[j], array[j+1]);
				swapped = true;
			}
		}
		if(swapped==false){
			break;
		}
	}
	return array;
}
